service/storage_proxy.cc

```diff
@@ -3210,6 +3222,25 @@ storage_proxy::create_write_response_handler(const std::tuple<lw_shared_ptr<paxo
                     inet_address_vector_topology_change(), inet_address_vector_topology_change(), std::move(tr_state), get_stats(), std::move(permit), std::monostate(), is_cancellable::no);
 }
 
+template<typename Range>
+bool storage_proxy::can_update_views(const Range& targets, const schema_ptr& s, tracing::trace_state_ptr tr_state) const {
+    if (!_db.local().find_column_family(s).views().empty() && targets.begin() != targets.end()) {
+        auto max_ep = gms::inet_address{};
+        auto max_backlog = db::view::update_backlog::no_backlog();
+        for (auto ep : targets) {
+            auto backlog = get_backlog_of(ep);
+            if (backlog > max_backlog) {
+                max_backlog = backlog;
+                max_ep = ep;
+            }
+        }
+        slogger.trace("Max view update backlog: {} on node {}", max_backlog.relative_size(), max_ep);
+        tracing::trace(tr_state, "Max view update backlog: {} on node {}", max_backlog.relative_size(), max_ep);
+        return max_backlog.relative_size() < 1;
```

Comment:
Nitpick: If always tracing the max backlog isn't important (is it?), a much simpler implementation would just replace all these 14 lines by just
```c++
for (const auto& ep : targets) {
    if (get_backlog_of(ep).relative_size() >= 1) {
        return false;
    }
}
return true;
```

Reply:
Looks good, I changed it